### DISCUSSION TITLE
Create a mod with embedded data pack?

### DISCUSSION THREAD

**Discussion by diskria** [FabricMC/fabric-api#2626] [Mod Dev Support] [answered]
Hello everyone. Tell me please how can I add a datapack to my client mod for single player game so that the datapack is automatically added to the world upon creation. 
As I understand it, everyone uses [this mod](https://www.curseforge.com/minecraft/mc-mods/drp-global-datapack) for this, but I don’t understand in which part of the code the datapack is added to the world in order to do the same in my mod. Also this mod requires the player to install the datapacks in the ".minecraft/global_packs" folder, but I want this to happen automatically. I need to add a datapack to the mod, but I'm not sure if it's even possible? My datapack needs to be extracted somewhere or it can work at the mod level? There are examples?

**Comment by warjort** [comment, marked as answer]
Your mod jar is automatically a datapack.
Fabric will add your mod's data subfolder to the "Fabric Mods" datapack.

e.g. this is the fabric api (which is itself just a mod) adding tags
https://github.com/FabricMC/fabric/tree/1.19.2/fabric-convention-tags-v1/src/generated/resources/data
They are in src/generated instead of src/main because they are created programmatically using datagen.

**Reply by diskria** [reply]
I moved the "data" folder from the datapack to the my mod resources and it really worked. Is this really enough? At least achievements from the datapack works...
